documentation/concepts/check.md

```diff
@@ -4,25 +4,88 @@ menu: check
+the `–changes` flag produces a report indicating differences between your target database (ie. the one you want to change) and what your migration scripts define (ie. the set of instructions you want to use to change your target database).  
+You can use this capability pre- and post-deployment: 
+- In pre-deployment scenarios to check the effect of your pending changes 
+- In post-deployment scenarios to capture a history of changes for retrospective auditing or reporting 
+
+In either scenario, using the `-changes` flag will help you infer which database objects will be/have been affected - and how – when you execute/have executed your migration script(s). 
+
+### Requirements and behaviour
+The `check –changes` command and flag works by building a temporary database. This temporary database is first made to reflect the state of your target schema, and then made to reflect your target schema with the pending changes applied.  
+
+The difference between the two states of this temporary database (target now, and target with changes applied) represents the effect your pending migrations will have (or have had) when the scripts are (or were) executed. This difference is captured as an artefact called a “Change Report”. The change report is available as both HTML (human readable) and JSON (machine readable) formats. 
+
+The process works like this: 
+
+1. Specify your target database location 
+    1. This is the database you want to apply your changes to, where Flyway is already being used to manage migrations (i.e. A Flyway migrations table exists)) 
+1. Specify a temporary database, either  
+    1. A location for the temporary database (Flyway will create the database for you), or 
+    1. An existing temporary database (note: Flyway will “clean” this database, so if you specify a full database, you must ensure it is ok to for Flyway to erase its schema) 
+1. Run `flyway check –changes` 
+
+Flyway’s `check –changes` will then: 
+1. Clean your temporary database 
+1. Check the latest version of your target database (let’s say it’s at V4) 
+1. Migrate the temporary database to the same version as your target database 
+1. Take a snapshot of the temporary database (now also at V4) 
+1. Migrate the temporary database up to the latest script versions in the project (let’s say it’s at V6) 
+1. Take a snapshot of the temporary database (now at V6) 
+1. Compare the V4 temporary database snapshot to the V6 temporary database snapshot 
+1. Generate a HTML (human readable) and JSON (machine readable) Change Report, indicating the additions, deletions, and modifications of database objects between V4 and V6 
+
+## Good things to know 
+- There is no requirement for the temporary database to be in your production system 
+- Please note that the temporary database **will be cleaned** before the operation starts 
+- The comparison engine is dependent on [.NET 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) which is why this is required. 
```

Comment:
It feels weird to introduce `the comparison engine` here without much context. Would it be better to reword to `The underlying comparison technology`?